docusaurus/docs/Android/04-compose/04-message-components/08-reactions-picker.mdx

```diff
@@ -0,0 +1,168 @@
+    onDismiss: () -> Unit = {},
+    ... // Content
+)
+```
+
+* `onMessageAction`: Handler used for triggering message actions such as **reply**, **edit**, **delete**, **react** and others.
+* `onDismiss`: Handler used when the component is dismissed by clicking outside of the component UI or pressing the system back button.
+
+```kotlin
+ReactionsPicker(
+    ..., // State
+    onMessageAction = { action ->
+        composerViewModel.performMessageAction(action)
+        listViewModel.performMessageAction(action)
+    },
+    onDismiss = { listViewModel.removeOverlay() },
+    ... // Content
+)
+```
+
+In the snippet above, you propagate the `action` to the `composerViewModel` and `listViewModel`, for them to store the latest action. This will update the UI accordingly.
+
+Alternatively, you call `listViewModel.removeOverlay()` to remove the overlay from the screen, in `onDismiss()`. It's important to note that `onMessageAction()` calls `removeOverlay()` internally, to hide the overlay.
+
+Next, let's see how to customize `ReactionsPicker`.
+
+## Customization
+
+`ReactionsPicker` allows you to customize the reactions you are showing:
+
+```kotlin
+@ExperimentalFoundationApi
+@Composable
+public fun ReactionsPicker(
+    ..., // State,
+    reactionTypes: Map<String, Int> = ChatTheme.reactionTypes,
+    ... // Actions and content
+)
+```
+
+* `reactionTypes`: Allows you customize which reactions are shown. By default it uses `reactionTypes` inside of [`ChatTheme`](../02-general-customization/01-chat-theme.mdx).
+
+You can customize these either by overriding `ChatTheme.reactionTypes` or passing in your own `Map` of reactions.
+
+By default `ReactionsPicker` looks like a bottom sheet, however you can customize it to look like a completely different component, such as a dialog, a drawer or whatever helps you retain the look and feel of your app.
+
+```kotlin
+ReactionsPicker(
+    modifier = Modifier
+        .align(Alignment.Center)
+        .padding(horizontal = 20.dp)
+        .wrapContentSize(),
+    shape = ChatTheme.shapes.attachment,
+    message = selectedMessage,
+    onMessageAction = { action ->
+        composerViewModel.performMessageAction(action)
+        listViewModel.performMessageAction(action)
+    },
+    onDismiss = { listViewModel.removeOverlay() },
+    cells = GridCells.Fixed(4)
+)
+```
+
+The code above will result in the following UI:
+
+![Custom ReactionsPicker Component](../../assets/compose_custom_reactions_picker_component.png)
+
+```kotlin
+@ExperimentalFoundationApi
+@Composable
+public fun ReactionsPicker(
+    ..., // State and actions
+    headerContent: @Composable ColumnScope.() -> Unit = {},
+    centerContent: @Composable ColumnScope.() -> Unit = {
+        DefaultReactionsPickerCenterContent(
+            message = message,
+            onMessageAction = onMessageAction,
+            cells = cells,
+            reactionTypes = reactionTypes
+        )
+    },
+)
+```
+
```

Comment:
here too: "Apart from customization through modifiers, the component provides these slots:"